pinot-common/src/main/java/org/apache/pinot/common/exception/QueryException.java

```diff
@@ -144,17 +153,29 @@ public static ProcessingException getException(ProcessingException processingExc
     return copiedProcessingException;
   }
 
-  public static String getTruncatedStackTrace(Exception exception) {
+  public static String getTruncatedStackTrace(Throwable exception) {
     StringWriter stringWriter = new StringWriter();
     exception.printStackTrace(new PrintWriter(stringWriter));
     String fullStackTrace = stringWriter.toString();
     String[] lines = fullStackTrace.split("\n");
```

Comment:
(nit) `StringUtils.split(fullStackTrace, '\n')` will be faster because no regex matching required

Reply:
I think it would be better not to do this by processing strings at ll and use the stack trace elements which will already have been filled in e.g.
```java
StringBuilder sb = new StringBuilder();
for (StackTraceElement frame : exception.getStackTrace()) {
  // maybe select frame
}
Throwable cause = exception.getCause();
if (cause != null) {
  for (StackTraceElement frame : cause.getStackTrace()) {
    // maybe select frame
  }
}
for (Throwable s : exception.getSuppressed()) {
   for (StackTraceElement frame : s.getStackTrace() {
     // maybe select frame
  } 
}
```

Reply:
Well, actually, I don't know if what I said is correct or not. It might be better, it might not, but I don't think it's important enough to find out.

Reply:
yeah I actually looked into this yesterday but seems like there are some indent and nesting logics buried underneath that I can call cause they are private. there for I opted to do this string processing instead.